apps/reaper/test/unit/reaper/event/event_handler_test.exs

```diff
@@ -162,24 +201,58 @@ defmodule Reaper.Event.EventHandlerTest do
     end
   end
 
+  describe "#{dataset_delete()}" do
+    @tag focus: true
+    test "should delete associated raw topic when dataset:delete event fires" do
+      dataset = TDG.create_dataset(id: "dataset_id", technical: %{sourceType: "ingest"})
+
+      non_matching_ingestion = TDG.create_ingestion(%{id: 1, targetDataset: "other_dataset"})
+      matching_ingestion = TDG.create_ingestion(%{id: 2, targetDataset: "dataset_id"})
+      another_matching_ingestion = TDG.create_ingestion(%{id: 3, targetDataset: "dataset_id"})
+
+      mock_view_state = %{
+        1 => %{
+          "ingestion" => non_matching_ingestion
+        },
+        2 => %{
+          "ingestion" => matching_ingestion
+        },
+        3 => %{
+          "ingestion" => another_matching_ingestion
+        }
+      }
+
+      allow(IngestionDelete.handle(any()), return: :ok)
+      allow(Brook.ViewState.get_all(@instance_name, :extractions), return: {:ok, mock_view_state})
```

Comment:
Spent a long time trying to get this to work with the `with_event` syntax to no avail. I'm not sure if it's got something against multiple merges within a test event, or if it was a red herring and the `:ok` thing was the culprit. Either way I think this is fine.